session-replay/Sources/DatadogSessionReplay/Drafts/SessionReplay.swift

```diff
@@ -14,11 +14,22 @@ public struct SessionReplay {
     public static func initialize(
         with configuration: SessionReplayConfiguration,
         in datadogInstance: DatadogCoreProtocol = defaultDatadogCore
-    ) throws -> SessionReplayController {
-        let sessionReplay = SessionReplayFeature(configuration: configuration)
-        try datadogInstance.register(feature: sessionReplay)
+    ) -> SessionReplayController {
+        do {
+            let feature = SessionReplayFeature(configuration: configuration)
+            try datadogInstance.register(feature: feature)
 
-        return sessionReplay
+            guard let scope = datadogInstance.scope(for: feature.name) else {
+                throw FatalError(description: "Failed to obtain `FeatureScope` for \(feature.name)")
+            }
+
+            feature.register(sessionReplayScope: scope)
```

Comment:
🥚 or 🐔 ? You need the scope in the Feature, but you need to register the Feature to get a scope. I think this should be solved by the core, but I don't have a simple solution 🤔 
I was expecting to keep ref to the core instance in Features, but then we would have a retain circle. Anyway, your usage is valid 👍

Reply:
Yes, it's exactly like you describe. From my point of view, this is okay for now 👍 - it doesn't produce much boilerplate nor extreme inconvenience. The main thing I dislike is that it enforces optionals in Feature implementation because we can't set things on initialization. This however is IMO the price we have to pay for having Features decoupled at the public API level:
```swift
Core.initialize()

Feature1.initialize()
Feature2.initialize() 
```
Coupling them together:
```swift
Core.initialize(with: [
   Feature1.initialize()
   Feature2.initialize() 
])
```
could be a way of injecting `featureScope` back to the Feature further in Core's `initialize()` implementation.

I like no of copuling more than I dislike optionals ⚖️ 🙂.